aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -226,8 +226,10 @@ impl AptosVM {
         &self,
         resolver: &'r S,
         session_id: SessionId,
+        transaction_context_opt: Option<TransactionContext>,
```

Comment:
I renamed `TransactionContext` to `UserTransactionContext`, to make things clearer. User transaction contexts are optional here because they may not be available in some other contexts, such as system transactions.